packages/bee-q/src/components/option-list/bq-option-list.tsx

```diff
@@ -0,0 +1,118 @@
+import { h, Component, Event, EventEmitter, Listen } from '@stencil/core';
+
+import { isHTMLElement } from '../../shared/utils';
+
+/**
+ * @part base - The component's internal wrapper.
+ */
+@Component({
+  tag: 'bq-option-list',
+  styleUrl: './scss/bq-option-list.scss',
+  shadow: true,
+})
+export class BqOptionList {
+  // Own Properties
+  // ====================
+
+  private baseElem: HTMLElement;
+  slot: HTMLSlotElement;
+
+  // Reference to host HTML element
+  // ===================================
+
+  // State() variables
+  // Inlined decorator, alphabetical order
+  // =======================================
+
+  // Public Property API
+  // ========================
+
+  // Prop lifecycle events
+  // =======================
+
+  // Events section
+  // Requires JSDocs for public API documentation
+  // ==============================================
+
+  /** Handler to be called when `bq-option` item loses focus. */
+  @Event() bqBlur: EventEmitter<HTMLBqOptionElement>;
+
+  /** Handler to be called when `bq-option` item gets focus. */
+  @Event() bqFocus: EventEmitter<HTMLBqOptionElement>;
+
+  /** Handler to be called when `bq-option` is selected (on click/enter press). */
+  @Event() bqSelect: EventEmitter<HTMLBqOptionElement>;
+
+  // Component lifecycle events
+  // Ordered by their natural call order
+  // =====================================
+
+  // Listeners
+  // ==============
+
+  @Listen('bqOptionBlur', { passive: true })
+  onBqOptionBlur(event: CustomEvent<HTMLElement>) {
+    const { target: item } = event;
+    if (isHTMLElement(item, 'bq-option')) this.bqBlur.emit(item);
+  }
+
+  @Listen('bqOptionFocus', { passive: true })
+  onBqOptionFocus(event: CustomEvent<HTMLElement>) {
+    const { target: item } = event;
+    if (isHTMLElement(item, 'bq-option')) this.bqFocus.emit(item);
+  }
```

Comment:
I don't think we need to emit blur and focus on the list wrapper, emitting `bqSelect` will be enough for now. I would recommend adding to the `bqSelect` the item value as well:

```tsx
  /** Handler to be called when `bq-option` is selected (on click/enter press). */
  @Event() bqSelect: EventEmitter<{ value: string; item: HTMLBqOptionElement }>;

...

  @Listen('bqClick', { passive: true })
  @Listen('bqEnter', { passive: true })
  onBqSelect(event: CustomEvent<HTMLElement>) {
    const { target: item } = event;

    this.optionItems.forEach((option: HTMLBqOptionElement) => (option.selected = option === item));
    if (isHTMLElement(item, 'bq-option')) {
      this.bqSelect.emit({ item, value: item.value });
    }
  }
```